src/main/java/org/craftercms/studio/api/v2/dal/PublishRequest.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright (C) 2007-2019 Crafter Software Corporation. All Rights Reserved.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.craftercms.studio.api.v2.dal;
+
+import java.io.Serializable;
+import java.time.ZonedDateTime;
+
+/**
+ * @author Dejan Brkic
+ */
+public class PublishRequest implements Serializable {
+
+    private static final long serialVersionUID = 1287567542272432148L;
+
+    private long id;
+    private String site;
+    private String environment;
+    private String path;
+    private String oldPath;
+    private String user;
+    private ZonedDateTime scheduledDate;
+    private String state;
+    private String action;
+    private String contentTypeClass;
+    private String submissionComment;
+    private String commitId;
+    private String packageId;
+
+    public class State {
+        public final static String READY_FOR_LIVE = "READY_FOR_LIVE";
+        public final static String PROCESSING = "PROCESSING";
+        public final static String COMPLETED = "COMPLETED";
+        public final static String CANCELLED = "CANCELLED";
+        public final static String BLOCKED = "BLOCKED";
+    }
+
+    public class Action {
+        public final static String NEW = "NEW";
```

Comment:
Should we name this `CREATE`?

Reply:
That is port from v1. Per naming convention established long time ago, it declares the file as new (not existing on the delivery node). If changed it will break DB.